### PR TITLE
Add ui overlay & restructure DOM with layered grids

### PR DESCRIPTION
**Pull request by dev-bnguyen**
We now have 2 main `div`s/containers we'll be calling 'Layers'

1. UI LAYER
      - Contains things on top (everything in this layer is `z-index`ed to '99')
      - All things interactive (clickable)
      - i.e. buttons, menus, boss hp bar (non-interactive, but needs to sit on top of UI)
      
2. ASSET LAYER
      - Things behind
      - Non-interactive
      - i.e. player, mobs, details
      - (player & mob are **not** interactive, it is the `grid-area` we define in our UI layer that links to our attack() function, which makes it interactive)

These layers are built using a 3x6 grid:

image

(See comments below to find out where they start)

Notes:
- Fixes 'BOSS' text from pushing down elements
- Fixes `Menu` to be in front of mob
- Always keeps `Player` and `Mob` on platform
- Re-organised css to mirror HTML from top to bottom

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -22,47 +22,68 @@ <h1>JavaScriptStory</h1>
...
+                <div id="ui-bottom">
+                    <div id="boss-details">
+                        <div class="boss-grid">
+                            <div id="boss-image"></div>
+                            <div id="progress-boss-hpbar"><div id="current-boss-hp-progress"></div></div>
+                        </div>
+                    </div>
+
+                    <button onClick="hax()">spam4hax</button>
+                </div>
```
Comment (dev-bnguyen) [inline index.html:56]: UI LAYER

File: index.html [hunk 2]
```diff
@@ -22,47 +22,68 @@ <h1>JavaScriptStory</h1>
...
-        </div>
 
-        <div id="boss-details">
-            <div class="boss-grid">
-                <div id="boss-image"></div>
-                <div id="progress-boss-hpbar"><div id="current-boss-hp-progress"></div></div>
+            <div id="asset-platform">
+                <img id="platform" />
             </div>
         </div>
```
Comment (dev-bnguyen) [inline index.html:86]: ASSET LAYER

File: style.css [hunk 3]
```diff
@@ -88,19 +86,71 @@ body {
...
+    grid-template-areas: 
+        "lvl . settings"
+        "clickable clickable clickable"
+        "clickable clickable clickable"
+        "clickable clickable clickable"
+        "bottom bottom bottom"
+        "bottom bottom bottom";
+    width: 100%;
+    height: 100%;
+}
```
Comment (dev-bnguyen) [inline style.css:144]: UI LAYER 3x6 grid

File: style.css [hunk 4]
```diff
@@ -141,17 +191,90 @@ body {
...
+    grid-template-areas: 
+        ". . ."
+        ". details ."
+        "player mob mob"
+        "player mob mob"
+        "platform platform platform"
+        ". . .";
+    width: 100%;
+    height: 100%;
+}
```
Comment (dev-bnguyen) [inline style.css:275]: ASSET LAYER 3x6 grid

**Review by HarryPdot — APPROVED** [review]

**Pull request merged**